### PR TITLE
allow PHP 7.3

### PR DESCRIPTION
**Pull request by mimmi20**
This PR intends to allow PHP 7.3 for this library and the use of build stages on travis.

This PR may replace PR #30.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,18 +3,55 @@ language: php
...
 
-after_script:
-  - wget https://scrutinizer-ci.com/ocular.phar; php ocular.phar code-coverage:upload --format=php-clover clover.xml
+install: travis_retry composer update --optimize-autoloader --prefer-dist --prefer-stable --no-progress --no-interaction --no-suggest $COMPOSER_FLAGS -vv
+
+script: vendor/bin/phpunit --colors --no-coverage
+
+jobs:
+  allow_failures:
+    - php: 7.3
```
Comment (Ocramius) [inline .travis.yml]: Allowing failures on 7.3 is probably not OK

File: composer.json [hunk 2]
```diff
@@ -3,16 +3,16 @@
     "description": "Doctrine Cache adapter for PSR-16 Simple Cache",
     "type": "library",
     "require": {
-        "php": "~7.1.0 || ~7.2.0",
+        "php": "^7.1",
```
Comment (Ocramius) [inline composer.json]: Should restrict to below 7.4

File: composer.json [hunk 3]
```diff
@@ -3,16 +3,16 @@
...
         "psr/simple-cache": "^1.0"
     },
     "require-dev": {
-        "cache/integration-tests": "dev-master#ff86c07073a5d6b7a7fe9019dd4f6d8fdaaca5ea",
-        "cache/tag-interop": "^1.0@dev",
-        "phpunit/phpunit": "^6.5",
-        "infection/infection": "^0.9.2",
-        "symfony/console": "^3.4"
+        "cache/integration-tests": "dev-master",
+        "cache/tag-interop": "dev-master",
```
Comment (Ocramius) [inline composer.json]: Please lock to a specific commit if using master

File: composer.json [hunk 4]
```diff
@@ -37,5 +37,12 @@
     },
     "provide": {
         "psr/simple-cache-implementation": "1.0"
-    }
+    },
+    "config": {
+        "optimize-autoloader": true,
```
Comment (Ocramius) [inline composer.json]: Let's remove this: it obfuscates autoloader issues during testing

Comment (mimmi20) [reply composer.json]: removed

File: .travis.yml (hunk 1 above)
Comment (mimmi20) [reply .travis.yml]: solved

File: composer.json (hunk 2 above)
Comment (mimmi20) [reply composer.json]: changed

File: composer.json (hunk 3 above)
Comment (mimmi20) [reply composer.json]: changed

File: .gitignore [hunk 5]
```diff
@@ -3,3 +3,5 @@ composer.lock
 phpunit.xml
 infection.json
 infection-log.txt
+.idea/
```
Comment (Ocramius) [inline .gitignore]: This should be in your [global `.gitignore`](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore)

File: .travis.yml [hunk 6]
```diff
@@ -3,18 +3,54 @@ language: php
...
+stages:
+  - test
+  - test with coverage
+  - infection
 
-script:
-  - vendor/bin/phpunit --coverage-text --coverage-clover=clover.xml --colors
-  - vendor/bin/infection --log-verbosity=none --threads=4
+before_install:
+  - stty cols 130
```
Comment (Ocramius) [inline .travis.yml]: What's the reason for a fixed column setting?

File: .travis.yml [hunk 7]
```diff
@@ -3,18 +3,54 @@ language: php
...
+    - php: nightly
+      env: COMPOSER_FLAGS="--ignore-platform-reqs"
+
+    - stage: test with coverage
+      os: linux
+      php: 7.1
+      env:
+        - COVERAGE=yes
+        - COMPOSER_FLAGS=""
+      before_install:
```
Comment (Ocramius) [inline .travis.yml]: Redundant with `before_install`

File: .travis.yml [hunk 8]
```diff
@@ -3,18 +3,54 @@ language: php
...
+        - stty cols 130
+        - composer self-update
```
Comment (Ocramius) [inline .travis.yml]: This is done by travis implicitly since a few months, AFAIK

File: src/SimpleCacheAdapter.php [hunk 9]
```diff
@@ -203,6 +204,11 @@ public function has($key) : bool
         return $this->doctrineCache->contains($key);
     }
 
+    /**
+     * @param \DateInterval $ttl
```
Comment (Ocramius) [inline src/SimpleCacheAdapter.php]: `@param` and `@return` are redundant with the type declaration, and can be dropped

File: .travis.yml (hunk 6 above)
Comment (mimmi20) [reply .travis.yml]: removed

File: .travis.yml (hunk 7 above)
Comment (mimmi20) [reply .travis.yml]: changed

File: .travis.yml (hunk 8 above)
Comment (mimmi20) [reply .travis.yml]: removed

File: src/SimpleCacheAdapter.php (hunk 9 above)
Comment (mimmi20) [reply src/SimpleCacheAdapter.php]: removed

**Pull request merged**